pkg/endpoint/endpoint.go

```diff
@@ -2007,3 +2007,106 @@ func (e *Endpoint) Delete(monitor monitorOwner, ipam ipReleaser, manager endpoin
 
 	return errs
 }
+
+type ContainerStartFunc func()
+
+// Create handles the creation of an endpoint. Returns an error if the endpoint
+// was unable to be created, or if the specified context's deadline is reached.
+// If syncBuild is true, this function waits for specific conditions until
+// returning:
+// * if the endpoint has a sidecar proxy, it waits for the endpoint's BPF
+// program to be generated for the first time.
+// * otherwise, waits for the endpoint to complete its first full regeneration.
+func (e *Endpoint) Create(ctx context.Context, cfunc ContainerStartFunc, syncBuild bool) error {
```

Comment:
can you specify a better name than `Create`? If the endpoint is not created then how come it has a method of itself? :boom:

Reply:
Done.